src/loaders/earthEngineLoader.js

```diff
@@ -84,48 +84,38 @@ const earthEngineLoader = async (config) => {
         // From database as favorite
         layerConfig = JSON.parse(config.config)
 
-        const { filter, params } = layerConfig
-
-        // Backward compability for layers saved before 2.41
-        if (filter) {
-            layerConfig.period = getPeriodFromFilter(filter, layerConfig.id)
-            delete layerConfig.filter
-        }
-
-        // Backward compability for layers saved before 2.41
-        if (params) {
-            layerConfig.style = params
-            if (params.palette) {
-                layerConfig.style.palette = params.palette.split(',')
-            }
-            delete layerConfig.params
-        }
-
         // Backward compability for layers with periods saved before 2.36
-        // (could also be fixed in a db update script)
         if (layerConfig.image) {
             const filter = layerConfig.filter?.[0]
 
             if (filter) {
-                const period = filter.arguments?.[1]
-                let name = String(layerConfig.image)
-
-                if (typeof period === 'string' && period.length > 4) {
-                    const year = period.substring(0, 4)
-                    filter.year = parseInt(year, 10)
+                const id = filter.arguments?.[1]
+                const name = String(layerConfig.image)
+                const year =
+                    typeof id === 'string' && id.length > 4
+                        ? parseInt(id.substring(0, 4), 10)
+                        : undefined
 
-                    if (name.slice(-4) === year) {
-                        name = name.slice(0, -4)
-                    }
-                }
+                layerConfig.period = { id, name, year }
 
-                filter.name = name
+                delete layerConfig.filter
             }
+            delete layerConfig.image
+        } else if (layerConfig.filter) {
+            // Backward compability for layers saved before v100.6.0
+            layerConfig.period = getPeriodFromFilter(filter, layerConfig.id)
+            delete layerConfig.filter
         }
 
-        // Backward compability for layers saved before 2.40
-        if (typeof layerConfig.params?.palette === 'string') {
-            layerConfig.params.palette = layerConfig.params.palette.split(',')
+        // Backward compability for layers saved before v100.6.0
+        if (layerConfig.params) {
+            layerConfig.style = layerConfig.params
+            // Backward compability for layers saved before 2.40
```

Comment:
Is this comment correct? Because this block of code splitting palette into an array was added in the big ee refactor PR. My guess is that the comment can be removed since it would actually be the same comment as on line 110?